### DISCUSSION TITLE
User Manual

### DISCUSSION THREAD

**Discussion by kbengs** [pdfarranger/pdfarranger#1051] [General]
Version 2.0
## Supported features
* Open one or several PDF documents
* Visually see all PDF pages side by side
* Zoom in/out
* Reorder PDF pages by drag and drop & copy and paste
* Reverse order
* Move or copy pages between instances of PDF Arranger
* Rotate pages (in 90 degree steps)
* Delete pages
* Duplicate pages
* Add blank pages
* Crop white borders (automatically)
* Crop margins (by dragging markers with mouse)
* Hide margins (by dragging markers with mouse)
* Resize pages (in % or mm)
* Generate booklet
* Import images (convert image to PDF)
* Extract text or image from a page to clipboard
* Explode into images (add all images in page as new separate pages)
* Split pages
* Merge pages (to one sheet)
* Add margins (with merge pages feature)
* Paste page as overlay or underlay onto another page
* Paste as odd or even pages
* Select pages (all, odd, even, same file, same format, invert, deselect)
* Edit metadata
* Save all to a PDF file
* Export some of the pages to one or more PDF documents
* Undo and redo
* Print
* Fullscreen and Zoom fit
* Light or dark theme
* Translated into several languages

## General

**Open PDF documents**
* Open: Opens the files in a new empty instance
* Import: Append the pages to the end of current instance
* Drag and drop files from a file manager
* Copy and paste files from a file manager

**Add pages**
* Drag and drop pages from another instance
* Copy and paste (also between instances)

**Image support**
* Requires img2pdf
* Added images are converted to PDF pages
* img2pdf >= 0.4.2 supports transparent images

**Save**
* Saves all to a PDF file

**Export**
* Save some of the pages to one or more PDF files
* A number at the end of the filename will be incremented at every export

**Scroll horizontally**
* Shift + scroll mouse wheel (when page can’t fit window horizontally)

**Pan view**
* Click and hold mouse wheel and drag mouse

**Zoom**
* Ctrl while scrolling mouse wheel
* Click zoom in/out on headerbar
* Keyboard +/-

**Toggle Zoom fit**
* Double-click a page or press key f

**Toggle fullscreen**
* F11

**Select pages**
* Click on a page, then hold shift and click on another page. The page range will be selected.
* Hold shift and navigate with keyboard (left, right, up, down, page up, page down, home, end)
* Click between pages and drag mouse. The range will be selected.
* Ctrl + a selects all. Ctrl + shift + a deselects all

**Add margins**
* Merge pages, set columns and rows to 1, set margin in mm

## Customization of keyboard shortcuts

You can change the default keyboard shortcuts. To do so, set `enable_custom` to `true` and edit the shortcuts in `config.ini` in the `[accelerators]` section.

*   For Linux: `~/.config/pdfarranger/config.ini`
*   For Windows installer version: `C:\Users\username\AppData\Roaming\pdfarranger\config.ini`
*   For Windows portable version, the file is located in the same folder as pdfarranger.exe. Optionally, this file can be removed. The application will then use the same config file location as in the installer version.

## Set custom language

Language can be selected in the `Preferences` dialog. (PDF Arranger > 1.9.2)

## Preserving of outlines and links
* Requires pikepdf >= 8.0
* Can be preserved for first opened document except for the following actions
  * Page size (changed), if page has links
    (pages which are destination for outlines or links can have page size changed)
  * Generate booklet
  * Merge pages
  * Hide margins
  * Paste as Overlay and underlay

**Comment by kbengs** [comment]
What do you think about this "document", if/how should we link to it from the app?
One motivation is that I think we should describe more in detail when links and outlines are lost.

**Reply by dreua** [reply]
Good idea to write something link this up, thanks!

I hope it is okay to just dump some thoughts, I can expand on them if necessary.

Formatting: My eyes don't like colons ":" at the end of each heading, bold formatting is enough imo.

What is the goal/audience here? Do we want all features for purposes of testing or do we want this as a user guide? (For completeness I'd add "create empty page" to the add pages section, for users guidance I'd try to keep it shorter and rather explain more.)

Missing features:
- Drop/Open/Import Images
- Select even/odd
- Undo/redo
- Do we want a list of all keyboard shortcuts? (Having them in app accessible with Ctrl+? as other Gnome apps might be even preferrable.)

The "Scroll" section has some very obvious points, mentioning the vertical scrolling makes sense however, as this is not obvious. (Again if this is aiming for a complete test suite that is a good list, it just depends on the goals.)

If we choose to link to it anywhere I'd convert this to a wiki page first, linking there feels more natural to me than linking to a discussion thread.

**Reply by dreua** [reply]
Another missing feature:
- export of graphics

**Reply by dreua** [reply]
We can also host is as a markdown file if that is preferred over the wiki. Either allows you to link to specific subheadings which would be very useful imo.

**Comment by kbengs** [comment]
Thanks for comments. I updated according to some of your points.

I'd see it most as a user manual. I think someone else than me could make this look more like a normal manual so feel free to edit/add text to the "User manual" post (or create a new post). Yes we could move it to the wiki later.

In "Content loss wawning" dialog we maybe could have a "See User Manual for more info" link which would go to the wiki. And maybe also a link in "About dialog".

**Reply by dreua** [reply]
Sounds all good to me. Note that GH discussions allows you to answer to "threads" vs creating a new thread which you did here.
